In [11]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--headless=new")#브라우저 슬립 패스
options.add_argument("window-size=1920x1080")
# options.add_argument("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "http://davelee-fun.github.io/"
driver.get(url)

element = driver.find_element(By.TAG_NAME, "h1")
print(element.text)

time.sleep(3)

driver.quit()

Teddy의 선물 블로그


#### 1. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 부모요소의 id 속성값이 dev_course_list인 자식요소의 class가 course인 데이터만 출력해주세요.

In [51]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--headless=new")#브라우저 슬립 패스
options.add_argument("window-size=1920x1080")
# options.add_argument("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://crawlingtest.netlify.app/"
driver.get(url)

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

# for element in elements :
#     print(element.text.strip().replace("  ", ""))

time.sleep(3)

driver.quit()

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 2. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 해당 사이트 내 커리큘럼 일정 표 안의 모든 데이터를 가져오세요.

In [36]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--headless=new")#브라우저 슬립 패스
options.add_argument("window-size=1920x1080")
# options.add_argument("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://crawlingtest.netlify.app/"
driver.get(url)

# elements = driver.find_elements(By.TAG_NAME, "td")
elements = driver.find_elements(By.CSS_SELECTOR, "table tr td")

# for element in elements :
#     print(element.text.strip().replace("  ", ""))

ele_list = []
for element in elements :
    ele_list.append(element.text)

print(ele_list[0], ele_list[1], ele_list[2])
print(ele_list[3], ele_list[4], ele_list[5])
print(ele_list[6], ele_list[7], ele_list[8])
    
time.sleep(3)

driver.quit()

일정 커리큘럼 타이틀 난이도
5.1 ~ 6.15 나만의 엣지있는 블로그 사이트 만들기 (취미로 익히는 IT) 초급
6.16 ~ 7.31 파이썬과 데이터과학 첫걸음 (IT 기본기 익히기) 난이도


#### 3. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 아래 데이터만 추출해오세요!!
- (중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]

In [49]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--headless=new")#브라우저 슬립 패스
options.add_argument("window-size=1920x1080")
# options.add_argument("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://crawlingtest.netlify.app/"
driver.get(url)

# elements = driver.find_element(By.CSS_SELECTOR, "#dev_course_list > li:nth-child(9) > a")

# print(elements.text)

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

print(elements[8].text)#리스트 형태에서 8번째 텍스트만 출력해라

#for문을 사용해서 위 값을 가져오는데 가져오는 값중 (중급)이 포한되어 있는 데이터만 출력해라
# for element in elements :
#     if "(중급)" in element.text :
#         print(element.text.strip().replace("  ", ""))

time.sleep(3)

driver.quit()

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 5. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- news 클래스 의 값만 찾아오세요

In [73]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("headless")#브라우저 슬립 패스
options.add_argument("window-size=1920x1080")
options.add_argument("--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://divjason.github.io/sellenium-test/"
driver.get(url)

# element = driver.find_element(By.TAG_NAME, "span")
element = driver.find_element(By.CSS_SELECTOR, "span.news")

print(element.text)

time.sleep(3)

driver.quit()

뉴스: 머신러닝 강좌가 나왔어요~~~


#### 6. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 보여지는 input tag 요소에 id 값으로 error@error.com password 값으로 1234 입력 후 로그인 버튼 자동으로 클릭하기
- 클릭 이후 이동한 페이지에서 message 라는 클래스 속성겂을 가지고 있는 요소의 텍스트 정보를 출력!!

In [90]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://divjason.github.io/sellenium-test/"
driver.get(url)

element = driver.find_element(By.NAME, "username")
element.clear()
element.send_keys("error@error.com")

element = driver.find_element(By.NAME, "password")
element.clear()
element.send_keys("1234")

# element.send_keys(Keys.RETURN)
element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")

element.click()

elements = driver.find_elements(By.CSS_SELECTOR, "div.message")

time.sleep(3)

for element in elements :
    print(element.text)

time.sleep(3)

# driver.save_screenshot("shot1.png")

driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~


#### 6. 다음 웹사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 보여지는 input tag 요소에 id 값으로 error@error.com password 값으로 1234 입력 후 로그인 버튼 자동으로 클릭하기
- 클릭 이후 이동한 페이지에서 message 라는 클래스 속성겂을 가지고 있는 요소의 텍스트 정보를 출력!!
- 해당 페이지에서 로그아웃 버튼을 클릭 후 이동한 페이지에서 다시 message 라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력

In [106]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui

from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup

service = Service(r"C:\Users\user\Downloads\디지털 마케팅\chromedriver\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)

url = "https://divjason.github.io/sellenium-test/"
driver.get(url)

element = driver.find_element(By.NAME, "username")
element.clear()
element.send_keys("error@error.com")

element = driver.find_element(By.NAME, "password")
element.clear()
element.send_keys("")

# element.send_keys(Keys.RETURN)
element_login = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")

element_login.click()

time.sleep(3)

element_logout = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")

element_logout.click()

time.sleep(3)

elements = driver.find_elements(By.CSS_SELECTOR, "div.message")

for element in elements :
    print(element.text)

# driver.save_screenshot("shot1.png")

driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
뉴스: 머신러닝 강좌가 나왔어요~~~
